### DISCUSSION TITLE
8bitdo PS Classic DIY Support

### DISCUSSION THREAD

**Discussion by tomdominer** [darthcloud/BlueRetro#159] [Q&A] [answered]
Like, others have said, thank you for this amazing project. Already built a homemade N64 adapter for the new online controllers and it worked a charm off the bat.

So, I've just bought a Playstation 1/2 adapter from [willsconsolemodifications.co.uk](https://www.willsconsolemodifications.co.uk/index.php?route=product/product&path=59_67&product_id=52). I tried to pair my [8bitdo diy PlayStation Classic Controller](https://download.8bitdo.com/Manual/Mod-Kit/Mod-kit-for-original-playstation-classic-controller.pdf) to the blueretro and, unfortunately, the d-pad isn't mapped, even if I put it in "X Input" mode.

The buttons, including shoulder buttons, work, but the d-pad doesn't do anything.

Happy to attempt to map the controls myself using the blueretro.io but don't know where to start?

Tom.

**Comment by darthcloud** [comment, marked as answer]
I would bet they mapped the dpad as a joystick, if you try a game that enable DS joystick by default does it work? Unfortunately the third middle button is what enable analog (PS buttons on PS3/4/5) which you don't have with this controller.

In the advance config -> mapping section, you can change the 4 first mapping which are the joystick ones (LX_\<direction\> LY_\<direction\> and map destination to LD-\<direction\> then click save.

**Reply by tomdominer** [reply]
Excellent assumption, worked a charm.

If I use another controller on here, will I have to reset the mapping?

**Reply by darthcloud** [reply]
Yes if you need left joystick to work.

**Comment by maniaco007** [comment]
![image](https://user-images.githubusercontent.com/82119174/188312758-4b3046c6-7cb6-49f6-acd8-207345c3a75d.png)
Please, could you help me?

I installed the 8Bitdo P30 kit, and I'm using BlueRetro Internally on my PSX Fat.

I made the change as described to make DPad work, but I was not successful, I am sending a capture of how the mapping is.

**Comment by thanius** [comment]
Hi, I too have this controller with same issue. However, after configuring as described, D-pad up and down works as described but the right directional is stuck. Hard to say if left is working as it's cancelling the right directional when pressed.
Should I need to do more changes with thresholds and whatnot? I'm on firmware 1.7.3 and using X-input

**Comment by darthcloud** [comment]
Did you connect in X input mode?

**Reply by thanius** [reply]
Hi, I was in X-input but had to resync, now it's working! Thanks

**Reply by maniaco007** [reply]
> Did you connect in X input mode?

I tried again, but without success.

The P30 kit works normally on Windows, Android and Nintendo Switch.

Through ESP32 on PSX, the Playstation 4 controller works correctly.

**Reply by thanius** [reply]
Ok, it was apparently a fluke that it worked that one time. I noticed the first directional being pressed is the one that gets stuck, the others work fine. Any suggestions?

**Reply by thanius** [reply]
Ok, so the issue is that the 8bitdo board doesn't save last used input mode, I have to force it into X-input or Android modes. 
I wonder if it defaults to Switch mode, if so, is there a way to make it work without buttons getting stuck?

**Comment by darthcloud** [comment]
Try upgrading to latest 8bitdo FW?

8bitdo switch mode doesn't work currently with BlueRetro.

**Reply by thanius** [reply]
Going to see if it possible!

**Comment by maniaco007** [comment]
Sorry, I got confused with the models.
My kit is the P30, for conventional control, not classic mini.
There is no fw update.

**Reply by vicsidious** [reply]
Got a solution for you! 
With every DIY kit from 8bitdo, you can "remap" the dpad to be either the left joystick, right joystick or dpad. 
Unfortunately, the instruction leaflet on this particular DIY kit doesn't state it BUT, if you look on their support page, there's a Q/A section that answers the question! Basically, BEFORE connecting the controller, press Up + Select for about 3 seconds. Yes, they are Mississipied-seconds! The top LED should quickly flash red and the DPAD should work! 

![image](https://user-images.githubusercontent.com/115555660/195646938-0e961274-c18c-44ea-8897-192030ba9b4c.png)

The support page is here: https://support.8bitdo.com/faq/ps-classic-kit-and-ps-original-kit.html
Hope it helps :)
-David

**Reply by thanius** [reply]
THIS is the correct answer! Worked perfectly and I can start the controller without holding in any buttons and it just works after the UP+SELECT combo

**Reply by vicsidious** [reply]
Glad it helped @thanius :)
